backend/models/role.go

```diff
@@ -2,7 +2,6 @@ package models
 
 type Role struct {
 	Common
-	Name string        `json:"name"`
-	Keys []*Credential `gorm:"many2many:key_role;"`
-	Door []*Door       `gorm:"many2many:role_door;"`
+	Name  string `json:"name"`
+	Doors []Door `json:"doors,omitempty" gorm:"many2many:role_door;"`
```

Comment:
What is there not more * (but the Door model still has it). Also why did you get rid of the credentials connection?

Reply:
Its unneccessary, the connection is still there if you look at Gorm https://gorm.io/docs/many_to_many.html for many to many we actually only need the definition in one table. The only benefit of having it as a back-reference is if you want to look both ways, but in this case we won't ever be looking from roles to keys just to figure out what roles a key has. Also the * seemed useless to me??? lmk if you had a specific reason, I should remove it from both sides though so its consistent

Reply:
For reference I left the Roles connection in doors, even though we aren't getting the role associations with a door cuz I could see us doing that in the future like a lil map with who has access to what door or smthn but even then its probably unneccessary, we really don't need the back-reference tbh

Reply:
Ended up adding it back in based on recent commit, although we don't need it for gets having it there so we can hard delete associations on deletion is nice (still don't see the point/value in the * here tho)